src/channel.js

```diff
@@ -0,0 +1,554 @@
+import {
+    arraysEqual,
+  } from './utils.js';
+import TileImage from 'ol/source/TileImage';
+import TileLayer from 'ol/layer/Tile';
+
+/** Extracts value of Pixel Spacing attribute from metadata.
+ *
+ * @param {object} metadata - Metadata of a DICOM VL Whole Slide Microscopy Image instance
+ * @returns {number[]} Spacing between pixel columns and rows in millimeter
+ * @private
+ */
+ function _getPixelSpacing(metadata) {
+    const functionalGroup = metadata.SharedFunctionalGroupsSequence[0];
+    const pixelMeasures = functionalGroup.PixelMeasuresSequence[0];
+    return pixelMeasures.PixelSpacing;
+  }
+
+/** BlendingInformation for DICOM VL Whole Slide Microscopy Image instances
+ * with Image Type VOLUME.
+ *
+ * @class
+ * @memberof channel
+ */  
+
+class BlendingInformation {
+  /*
+  * An interface class to set/get the visualization/presentation parameters from a channel object
+  * @param {string} opticalPathIdentifier, ID of the channel 
+  * @param {number[]} color
+  * @param {number} opacity
+  * @param {number[]} thresholdValues
+  * @param {boolean} visible
+  */
+  constructor(
+    opticalPathIdentifier, 
+    color,
+    opacity,
+    thresholdValues,
+    visible) {
+
+    this.opticalPathIdentifier = opticalPathIdentifier;
+    this.color = [...color];
+    this.opacity = opacity;
+    this.thresholdValues = [...thresholdValues];
+    this.visible = visible;
+    }
+}
+
+/** Channel for DICOM VL Whole Slide Microscopy Image instances
+ * with Image Type VOLUME.
+ *
+ * @class
+ * @memberof channel
+ */
+
+class Channel {
+/**
+ * Create a channel instances which contains all the visualization/presnetation parameters and OpenLayer objects for a Whole Slide Microscopy Image.
+ *
+ * @param {object} BlendingInformation
+ * 
+ * TO DO: implement API to select focal plane in the case of 3D channels (channels with a bandwith)
+ * TO DO: use DICOM attributes for loading/saving the channel parameters (i.e. load/save the presentation state/BlendingInformation in DICOM), for example:
+ *        [x] Select area for display: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.10.4.html 
+ *        [x] Clipping pixel values: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.33.html#table_C.11.33.1-1 
+ *        [x] Select channels for display and specify the color of each channel: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.34.html
+ *        [x] Blending of images: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.34.html http://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_N.2.6.html
+ * TO DO: for coloring monochrome channels we download the uncompressed binaries data. 
+ *        For a large N of channels, this could be a substantial bottleneck. 
+ *        Check performance downloading binary (octet-stream) vs downloading compressed (jpeg) + decoding. 
+ *        https://github.com/cornerstonejs/codecs 
```

Comment:
Shall we move some of the TODO items out of the source code and into issues?

Reply:
There already issues for all these To Do in the slim repository. I removed these TO DO from the source